examples/tina-self-hosted-demo/pages/api/tina/[...routes].ts

```diff
@@ -0,0 +1,16 @@
+import { TinaNodeBackend } from '@tinacms/datalayer/dist/node'
+import { TinaAuthJSOptions, NextAuthAuthentication } from 'tinacms-authjs'
+
+import databaseClient from '../../../tina/__generated__/databaseClient'
+
+export const nextAuthOptions = TinaAuthJSOptions({
+  databaseClient: databaseClient,
+  secret: process.env.NEXTAUTH_SECRET,
+})
+
+const tinaNodeBackend = TinaNodeBackend({
+  authentication: NextAuthAuthentication({ nextAuthOptions }),
+  databaseClient,
+})
+
+export default tinaNodeBackend
```

Comment:
See https://github.com/tinacms/tinacms/pull/4231#issuecomment-1747092012

Reply:
@kldavis4 all of the next backend specific features have been removed.  Are you talking about this comment https://github.com/tinacms/tinacms/pull/4231#issuecomment-1747386131 ?

Reply:
huh?

<img width="762" alt="Screen Shot 2023-10-05 at 13 55 04" src="https://github.com/tinacms/tinacms/assets/370955/fb74815e-7e25-469f-b17d-e8ae56ad6b72">

Reply:
The comment you linked to is

> I think this should be a node backend (not next) since it doesn't use any next specific features

I believe this is already done?

From the context it looked like you might be talking about the comment I linked to 

> I believe also James suggested we explicitly pass in the req, res objects the way the nextauth api handler [does](https://github.com/tinacms/tinacms/blob/feat/self-hosted-improvements/examples/tina-self-hosted-demo/pages/api/auth/%5B...nextauth%5D.ts#L5)


This can be done by doing 

```ts
const handler = TinaNodeBackend(
  // ...
)
export default (req, res) =>{
   handler(req, res)
}
``` 

IMO we should not do it this way since I think its a bit more cumbersome and not the standard that other libraries are doing. 

See examples of TRPC and NextAuth 
- https://github.com/nextauthjs/next-auth-example/blob/main/app/api/auth/%5B...nextauth%5D/route.ts
- https://github.com/trpc/examples-next-prisma-starter/blob/main/src/pages/api/trpc/%5Btrpc%5D.ts

But If @jamespohalloran and yourself wan't to be more explicit I don't have an issue with it.

Reply:
Here's the reference for this form of initialization in next-auth: https://next-auth.js.org/configuration/initialization#advanced-initialization

I think I prefer it in the more explicit form as we don't have to answer the question of how to modify the request object (or having a page like this to document it) but I'll defer to James if he thinks otherwise